dhcpv4/nclient4/client.go

```diff
@@ -521,6 +537,15 @@ var errDeadlineExceeded = errors.New("INTERNAL ERROR: deadline exceeded")
 // ClientHWAddr is returned.
 func (c *Client) SendAndRead(ctx context.Context, dest *net.UDPAddr, p *dhcpv4.DHCPv4, match Matcher) (*dhcpv4.DHCPv4, error) {
 	var response *dhcpv4.DHCPv4
+	//check if the request packet has all options required by c.clientIdOptions
```

Comment:
I would remove this check

Reply:
This check is more for preventing user to shoot itself in the foot, I agree it is not mandatory, I could remove it

Reply:
I think this is something for callers to do, not for this library, too.